src/main/java/org/springframework/integration/dsl/StandardIntegrationFlow.java

```diff
@@ -88,7 +85,9 @@ public void start() {
 	public void stop(Runnable callback) {
 		if (this.running) {
 			AggregatingCallback aggregatingCallback = new AggregatingCallback(this.lifecycles.size(), callback);
-			for (SmartLifecycle lifecycle : this.lifecycles) {
+			ListIterator<SmartLifecycle> iterator = this.lifecycles.listIterator(this.lifecycles.size());
+			while (iterator.hasPrevious()) {
```

Comment:
OK. I can add some JavaDocs to explain the start/stop with different directions.